### PR TITLE
webContents: providing header details with response event

### PR DESCRIPTION
**Pull request by deepak1556**
better approach to #1604 

Depends on https://github.com/atom/libchromiumcontent/pull/120

### CODE REVIEW

**Comment by maxogden** [comment]
just curious, what is an example value of `header`? is it going to be a newline separated list of all headers, e.g. like this?

```Cache-Control:no-cache\nContent-Encoding:gzip```

if so, shouldn't it be `headers` instead of `header`?

**Comment by deepak1556** [comment]
@maxogden Yup it will be new-line delimited string, named it header since it was one single string. Will change it to `headers`

```
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Date: Wed, 20 May 2015 15:45:11 GMT
Pragma: no-cache
Expires: Fri, 01 Jan 1990 00:00:00 GMT
Cache-Control: no-cache, no-store, must-revalidate
Last-Modified: Sun, 17 May 1998 03:00:00 GMT
X-Content-Type-Options: nosniff
Content-Type: image/gif
Server: Golfe2
Content-Length: 35
Alternate-Protocol: 443:quic,p=1
```

**Comment by sindresorhus** [comment]
Any reason `headers` is not an object, like `{ 'Access-Control-Allow-Origin': '*', ... }`? This would be consistent with [Node.js](https://nodejs.org/api/http.html#http_http). Lowercasing the keys would also be nice.

**Comment by deepak1556** [comment]
@sindresorhus no reason ws just lazy ;) have fixed it, now headers will be an object.

```
{"access-control-allow-origin":"*","alternate-protocol":"443:quic,p=1","cache-control":"no-cache, no-store, must-revalidate","content-length":"35","content-type":"image/gif","date":"thu, 21 may 2015 21:18:50 gmt","expires":"fri, 01 jan 1990 00:00:00 gmt","last-modified":"sun, 17 may 1998 03:00:00 gmt","pragma":"no-cache","server":"golfe2","x-content-type-options":"nosniff"}
```

**Comment by sindresorhus** [comment]
:+1:

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -309,13 +311,23 @@ void WebContents::DidStopLoading(content::RenderViewHost* render_view_host) {
 
 void WebContents::DidGetResourceResponseStart(
     const content::ResourceRequestDetails& details) {
+  net::HttpResponseHeaders* headers = details.headers.get();
+  base::DictionaryValue response_headers;
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: We can save the headers in `mate::Dictionary` so there is no need for converting `base::DictionaryValue` to `v8::Object`.

Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: :+1:  these callbacks are threadsafe right, I dont have to lock `v8::Isollate` before the dictionary creation ?

File: atom/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -309,13 +311,25 @@ void WebContents::DidStopLoading(content::RenderViewHost* render_view_host) {
 
 void WebContents::DidGetResourceResponseStart(
     const content::ResourceRequestDetails& details) {
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  mate::Dictionary response_headers(isolate, v8::Object::New(isolate));
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: You still need to add V8's lock and create a handle scope.

Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: Done, thanks!

**Comment by nistvan86** [comment]
Cool. :) As i see it's now in the release. Thank you!

**Pull request merged**